File: Serenity.Script.UI/FilterPanel/FilterDisplayBar.cs
using jQueryApi;
using System;
using System.Runtime.CompilerServices;

namespace Serenity
{
    [Imported(ObeysTypeSystem = true)]
    public class FilterDisplayBar : FilterWidgetBase<object>
    {
        public FilterDisplayBar(jQueryObject div)
            : base(div, null)
        {
            this.Element.Find(".cap").Text(Q.Text("Controls.FilterPanel.EffectiveFilter"));
            this.Element.Find(".edit").Text(Q.Text("Controls.FilterPanel.EditFilter"));
            this.Element.Find(".reset").Attribute("title", Q.Text("Controls.FilterPanel.ResetFilterHint"));

            jQueryEventHandler openFilterDialog = delegate(jQueryEvent e)
            {
                e.PreventDefault();
                var dialog = new FilterDialog();
                dialog.FilterPanel.Store = this.Store;
                dialog.DialogOpen();
            };

            this.Element.Find(".edit").Click(openFilterDialog);
            this.Element.Find(".txt").Click(openFilterDialog);

            this.Element.Find(".reset").Click(e =>
            {
                e.PreventDefault();
                this.Store.Items.Clear();
                this.Store.RaiseChanged();
            });
        }

        protected override void FilterStoreChanged()
        {
            base.FilterStoreChanged();

            var displayText = Store.DisplayText.TrimToNull();
            this.Element.Find(".current").Toggle(displayText != null);
            this.Element.Find(".reset").Toggle(displayText != null);
            this.Element.Find(".txt").Text("[" + (displayText ?? Q.Text("Controls.FilterPanel.EffectiveEmpty")) + "]");
        }

        protected override string GetTemplate()
        {
            return 
                "<div>" + 
                    "<a class='reset'></a>" +
                    "<a class='edit'></a>" + 
                    "<div class='current'>" + 
                        "<span class='cap'></span>" +
                        "<a class='txt'></a>" + 
                    "</div>" + 
                "</div>";
        }
    }
}

File: packages/corelib/src/ui/filtering/filterdisplaybar.ts
import { Fluent, localText } from "../../base";
import { Decorators } from "../../types/decorators";
import { FilterDialog } from "./filterdialog";
import { FilterWidgetBase } from "./filterwidgetbase";

@Decorators.registerClass('Serenity.FilterDisplayBar')
export class FilterDisplayBar<P = {}> extends FilterWidgetBase<P> {

    protected renderContents() {
        var openFilterDialog = (e: Event) => {
            e.preventDefault();
            var dialog = new FilterDialog({});
            dialog.get_filterPanel().set_store(this.get_store());
            dialog.dialogOpen(null);
        };

        return Fluent("div")
            .append(Fluent("a")
                .class("reset")
                .attr("title", localText('Controls.FilterPanel.ResetFilterHint'))
                .on("click", e => {
                    e.preventDefault();
                    this.get_store().get_items().length = 0;
                    this.get_store().raiseChanged();
                }))
            .append(Fluent("a")
                .class("edit")
                .text(localText('Controls.FilterPanel.EditFilter'))
                .on("click", openFilterDialog))
            .append(Fluent("div")
                .class("current")
                .append(Fluent("span")
                    .class("cap")
                    .text(localText('Controls.FilterPanel.EffectiveFilter')))
                .append(Fluent("a")
                    .class("txt")
                    .on("click", openFilterDialog))).getNode();
    }

    protected filterStoreChanged() {
        super.filterStoreChanged();

        var displayText = this.get_store().get_displayText()?.trim() || null;

        this.element.findFirst('.current').toggle(displayText != null);
        this.element.findFirst('.reset').toggle(displayText != null);

        if (displayText == null)
            displayText = localText('Controls.FilterPanel.EffectiveEmpty');

        this.element.findFirst('.txt').text('[' + displayText + ']');
    }
}
